Public/Src/FrontEnd/MsBuild/PipConstructor.cs

```diff
@@ -560,7 +560,7 @@ private bool TryAddMsBuildArguments(ProjectWithPredictions project, PipDataBuild
                     pipDataBuilder.Add(PipDataAtom.FromString($"/t:{target}"));
                 }
             }
-            else
+            else if (project.ProjectReferences.Count > 0)

```

Comment:
>if (project.ProjectReferences.Count > 0) [](start = 17, length = 40)

Not sure about this. Shouldn't all projects comply to the project reference protocol in this new world of static graph MSBuild?
I understand that when projects are leaves the protocol is inconsequential, but a project not specifying the protocol sounds fishy enough to me to do something about that (we warn today, but we may decide to error in the future). 
@micodoba@microsoft.com , thoughts?

Reply:
Why is it fishy? In CloudBuild, for example, there are many leaf projects that are just moving files around, or strong-signing NuGet dependencies; moreover, there are roughly 40 of these.

In cases like this, warning about missing the TRP for leaves is just adding noise, as there may be many other projects which genuinely need attention. Furthermore, the warnings are misleading: there are no references, so what should the TRP be?

Reply:
Is there an option to downgrade the warning to informational logs for leaf nodes?

Reply:
I didn't see any way to downgrade the warning, so I just added a new informational event specific for the leaves.

Reply:
In an ideal world all SDKs implement the protocol. This protocol specification goes beyond the concrete references a node may have. So even if a node doesn't have any references, the protocol will say what happens if references are added.

But after discussing this with Mihai, he convinced me that in our current not-ideal world, it is ok to let that slide. So I'm good.

---
In reply to: [281338766](https://github.com/Microsoft/BuildXL/pull/256#discussion_r281338766) [](ancestors = 281338766)